rclcpp/include/rclcpp/publisher_factory.hpp

```diff
@@ -75,21 +75,24 @@ struct PublisherFactory
 /// Return a PublisherFactory with functions setup for creating a PublisherT<MessageT, Alloc>.
 template<typename MessageT, typename Alloc, typename PublisherT>
 PublisherFactory
-create_publisher_factory(std::shared_ptr<Alloc> allocator)
+create_publisher_factory(
+  const PublisherEventCallbacks & event_callbacks,
+  std::shared_ptr<Alloc> allocator)
 {
   PublisherFactory factory;
 
   // factory function that creates a MessageT specific PublisherT
   factory.create_typed_publisher =
-    [allocator](
+    [event_callbacks, allocator](
```

Comment:
Just to double check: this is capturing by copy EventsCallback - did we plan in the design to make this copyable? Are we sure that it's safe to do so? (i.e. a proper in-depth copy is done or if not, we're sure objects won't be freed at some point?)